core/capabilities/remote/target/client.go

```diff
@@ -46,20 +51,44 @@ func NewClient(ctx context.Context, lggr logger.Logger, remoteCapabilityInfo com
 		messageIDToCallerRequest: make(map[string]*request.ClientRequest),
 	}
 
-	go func() {
-		ticker := time.NewTicker(requestTimeout)
-		defer ticker.Stop()
-		for {
-			select {
-			case <-ctx.Done():
-				return
-			case <-ticker.C:
-				c.expireRequests()
+	return c
+}
+
+func (c *client) Start(ctx context.Context) error {
```

Comment:
I'm making target client/server Service-compatible in my PR too (just copied StartOnce to match your implementation)